lib/Dialects/VectorExt/Transform/VectorWarpUtils.cpp

```diff
@@ -446,6 +437,83 @@ struct WarpOpTransferWrite : public OpRewritePattern<vector::TransferWriteOp> {
     return success();
   }
 
+  /// Extract TransferWriteOps of vector<1x> into a separate warp op.
+  /// Note: The new write op has a mask, so it is not further matched by this
+  /// pattern.
+  LogicalResult tryExtractOp(RewriterBase &rewriter,
+                             vector::TransferWriteOp writeOp,
+                             WarpSingleLaneOp warpOp) const {
+    Location loc = writeOp.getLoc();
+    VectorType vecType = writeOp.getVectorType();
+
+    // Only vector<1x> is supported at the moment.
+    if (vecType.getShape().size() != 1 || vecType.getShape()[0] != 1)
+      return failure();
+
+    SmallVector<Value> yieldValues = {writeOp.vector()};
+    SmallVector<Type> retTypes = {vecType};
+    WarpSingleLaneOp newWarpOp = moveRegionToNewWarpOpAndAppendReturns(
+        rewriter, warpOp, yieldValues, retTypes);
+    rewriter.setInsertionPointAfter(newWarpOp);
+
+    // Create a second warp op that contains only writeOp (and a mask). Pass in
+    // the vector that should be written.
+    auto secondWarpOp = rewriter.create<WarpSingleLaneOp>(
+        loc, TypeRange(), newWarpOp.laneid());
+    secondWarpOp.argsMutable().assign(
+        newWarpOp.getResult(newWarpOp.getNumResults() - 1));
+    Block &body = secondWarpOp.getBodyRegion().front();
+    VectorType broadcastedType =
```

Comment:
I don't think we need to do that since we know we broadcast the value from the previous region we can just use the value directly as a uniform defined outside of the region